specifyweb/stored_queries/execution.py

```diff
@@ -479,14 +479,16 @@ def build_query(session, collection, user, tableid, field_specs, recordsetid=Non
     will be filtered to items from the given record set unless None.
 
     replace_nulls = if True, replace null values with ""
+
+    distinct = if True, do not return record IDs and query distinct rows
     """
     model = models.models_by_tableid[tableid]
     id_field = getattr(model, model._id)
 
     query = QueryConstruct(
         collection=collection,
         objectformatter=ObjectFormatter(collection, user, replace_nulls),
-        query=session.query(id_field),
+        query=session.query() if distinct else session.query(id_field),
```

Comment:
I think this should be `session.query().distinct()` to actually add the DISTINCT clause to the SQL, so that the behavior matches the comment at line 483. This would take the responsibility for adding that clause over from the `execute` function.